scripts/deploy_docker.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+
+PROJECT_ID=lightstep
+DATETIME=$(date "+%Y-%m-%d_%H-%M-%S")
+BASENAME=lightstep-benchmark
+PROJECT_ID=$(gcloud config --format text list project | cut -d' ' -f2)
+
+NAME=gcr.io/${PROJECT_ID}/${BASENAME}
+
+docker build --no-cache -t ${NAME}:${DATETIME} ${GOPATH}/../docker
+docker tag -f ${NAME}:${DATETIME} ${NAME}:latest
+
+gcloud docker push ${NAME}:${DATETIME}
+gcloud docker push ${NAME}:latest
+
+# Some docker-machine magic to create a machine
+eval $(docker-machine env lightstep-benchmarks-1cpu)
+
+gcloud docker run -- -d ${NAME} ./controller
```

Comment:
I spent a couple of frustrated hours on this (and am probably doing this in
an unconventional way, since the documentation is very difficult), but it
does run remotely provided the docker-machine magic is done first. The
"run" uses a remote docker daemon on the GCE instance.

The machine was created thusly:
docker-machine create \
--driver google \
--google-project helpful-cat-xxx lightstep-benchmarks-vm

and then the eval $(docker machine env lightstep-benchmarks-vm) makes it
look like my GCE instance is local, and then gcloud run does the right
thing.
I'm still feeling confused, but it's running.

ᐧ

On Wed, Apr 20, 2016 at 12:56 PM, spoons <notifications@github.com> wrote:

> In scripts/deploy_docker.sh
> <https://github.com/lightstep/lightstep-benchmarks/pull/7#discussion_r60476406>
> :
>
> > +DATETIME=$(date "+%Y-%m-%d_%H-%M-%S")
> > +BASENAME=lightstep-benchmark
> > +PROJECT_ID=$(gcloud config --format text list project | cut -d' ' -f2)
> > +
> > +NAME=gcr.io/${PROJECT_ID}/${BASENAME} <http://gcr.io/$%7BPROJECT_ID%7D/$%7BBASENAME%7D>
> > +
> > +docker build --no-cache -t ${NAME}:${DATETIME} ${GOPATH}/../docker
> > +docker tag -f ${NAME}:${DATETIME} ${NAME}:latest
> > +
> > +gcloud docker push ${NAME}:${DATETIME}
> > +gcloud docker push ${NAME}:latest
> > +
> > +# Some docker-machine magic to create a machine
> > +eval $(docker-machine env lightstep-benchmarks-1cpu)
> > +
> > +gcloud docker run -- -d ${NAME} ./controller
>
> Likewise, you don't need (I think) the gcloud part unless you are pushing
> or pulling something from GCR.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightstep/lightstep-benchmarks/pull/7/files/400944af39a631945f55821d496d0ce04e607af9#r60476406>
>